<a href="https://colab.research.google.com/github/sureshora/MLOps/blob/master/Project_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Project:1 ** Sentiment Intensity Analyser

The **SentimentIntensityAnalyzer** class provides a method called polarity_scores() that takes **a piece of text as input** and returns a dictionary containing the sentiment scores for the text. The dictionary contains four keys: neg , neu , pos , and compound . neg : the negative sentiment score (between 0 and 1).

In [7]:
# Import NLTK and download the VADER lexicon
import nltk
nltk.download('vader_lexicon')
from nltk.sentiment import SentimentIntensityAnalyzer

[nltk_data] Downloading package vader_lexicon to /root/nltk_data...
[nltk_data]   Package vader_lexicon is already up-to-date!


In [8]:
# Create a sentiment analyzer instance
sia = SentimentIntensityAnalyzer()

# Analyze the sentiment of a text
text = "I loved the movie !!! It was amazing."
sentiment = sia.polarity_scores(text)

In [9]:
# Print the sentiment scores
print(sentiment)

{'neg': 0.0, 'neu': 0.368, 'pos': 0.632, 'compound': 0.8617}


OutPut
{'neg': 0.0, 'neu': 0.368, 'pos': 0.632, 'compound': 0.8617}
In this example, we use the SentimentIntensityAnalyzer class from the NLTK library to perform sentiment analysis on the given text. The polarity_scores() method returns a dictionary of sentiment scores, including the negative (neg), neutral (neu), positive (pos), and compound (compound) scores.
In the provided text, "I loved the movie! It was amazing," the sentiment analysis indicates a highly positive sentiment.

The** compound score of 0.8617 suggests a strong positive sentiment**, while the negative and neutral scores are close to zero, indicating minimal negativity or neutrality in the text.

